### PR TITLE
libmtp, libgphoto2, gphoto2: Switch to Github, allow building from repo

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

Github is where upstream's development happens now,
and building from repo makes adding patches and overriding the commit easier.

Also use pname+version, see #103997, and reformat for reduced merge conflicts.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nh2** [comment]
cc @jcumming

File: pkgs/development/libraries/libgphoto2/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgphoto2/default.nix:51]: What does the XXX: mean?

Comment (nh2) [reply pkgs/development/libraries/libgphoto2/default.nix:51]: It's [a way to indicate a sort of TODO or ugliness note](https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments), but I don't think it's good, exactly because it isn't obvious what it means.

Comment (jcumming) [reply pkgs/development/libraries/libgphoto2/default.nix:51]: I recognized the writing style as mine, but I didn't remember writing that comment. 

Git blame claims I wrote it 8 years ago!

The upstream project now reports: [GNU Lesser General Public License v2.1](https://github.com/gphoto/libgphoto2/blob/master/COPYING). Let's use that.

Comment (jtojnar) [reply pkgs/development/libraries/libgphoto2/default.nix:51]: @jcumming Unfortunately, the GitHub is not able to detect if the license is `*gpl-or-later` or `*gpl-only`, see https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

Comment (nh2) [reply pkgs/development/libraries/libgphoto2/default.nix:51]: I'll leave it to you people to figure that one out :grin: 

Beyond that, anybody mind approving if it looks good?

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**
